costmap_2d/src/costmap_layer.cpp

```diff
@@ -3,6 +3,40 @@
 namespace costmap_2d
 {
 
+void CostmapLayer::clearWindow(double min_x, double min_y, double max_x, double max_y)
+{
+  int start_x, start_y, end_x, end_y;
+  worldToMapNoBounds(min_x, min_y, start_x, start_y);
+  worldToMapNoBounds(max_x, max_y, end_x, end_y);
+
+  unsigned char* grid = getCharMap();
+  for(int x=0; x<(int)getSizeInCellsX(); x++){
```

Comment:
This clear window only happens when the costmap clear happens.  I'll make a future issue to further optimize it.